src/Core/Translate/IntrinsicTranslator.cs

```diff
@@ -168,6 +169,74 @@ CodeExpression Translate_sum(Application appl, CodeExpression[] args)
             return null;
         }
 
+        CodeExpression Translate_range(Application appl, CodeExpression[] args)
+        {
+            var argsCount = args.Length;
+
+            if (1 <= argsCount && argsCount <= 3)
+            {
+                m.EnsureImport("System");
+                m.EnsureImport("System.Linq");
+
+                var enumerableRange = m.MethodRef(m.TypeRefExpr("Enumerable"), "Range");
+                var mathCeiling = m.MethodRef(m.TypeRefExpr("Math"), "Ceiling");
+                var convertToInt32 = m.MethodRef(m.TypeRefExpr("Convert"), "ToInt32");
+                var convertToDouble = m.MethodRef(m.TypeRefExpr("Convert"), "ToDouble");
+
+                switch (argsCount)
+                {
+                    case 1:
+                    {
+                        // Enumerable.Range(0, count);
+
+                        var startExp = new CodePrimitiveExpression(0);
+                        var countExp = args[0];
+                        return m.Appl(enumerableRange, startExp, countExp);
+                    }
+                    case 2:
+                    {
+                        // Enumerable.Range(0, count);
+
+                        var startExp = args[0];
+                        var stopExp = args[1];
+                        var countExp = new CodeBinaryOperatorExpression(stopExp, CodeOperatorType.Sub, startExp);
+                        return m.Appl(enumerableRange, startExp, countExp);
+                    }
+                    case 3:
+                    {
+                        // Enumerable.Range(0, count).Select(x => start + x * step);
+
+                        var startExp = args[0];
+                        var stopExp = args[1];
+                        var stepExp = args[2];
+
+                        // count = (stop - start) divide_ceiling_by step;
+                        var rawCountExp = new CodeBinaryOperatorExpression(stopExp, CodeOperatorType.Sub, startExp);
+                        var rawDoubleCountExp = m.Appl(convertToDouble, rawCountExp);
+                        var countDividedExp = new CodeBinaryOperatorExpression(rawDoubleCountExp, CodeOperatorType.Div, stepExp);
+                        var countCeilingExp = m.Appl(mathCeiling, countDividedExp); 
+                        var countExp = m.Appl(convertToInt32, countCeilingExp);
+                        
+                        // Enumerable.Range(0, count);
+                        var rangeExp = m.Appl(enumerableRange, new CodePrimitiveExpression(0), countExp);
+
+                        // x => start + x * step;
+                        var lambdaArgExp = new CodeVariableReferenceExpression("x");
```

Comment:
If either `start` or `step` contain `x` in their expressions, this will generate incorrect code. Instead, use a generated symbol to avoid conflicts:
```
var lambdaArgExp = expTranslator.gensym.GenSymLocal("_p_", m.TypeRef("object"));
```